packages/core/src/store/state-store.ts

```diff
@@ -1,11 +1,80 @@
-import type { StreamState, StreamStateHolder } from '@ceramicnetwork/common'
-import type { StreamID } from '@ceramicnetwork/streamid'
-
-export interface StateStore {
-  open(networkName: string): Promise<void>
-  close(): Promise<void>
-  save(streamStateHolder: StreamStateHolder): Promise<void>
-  load(streamId: StreamID): Promise<StreamState | null>
-  list(streamId?: StreamID | null, limit?: number): Promise<string[]>
-  remove(streamId: StreamID): Promise<void>
+import { StreamID } from '@ceramicnetwork/streamid'
+import {
+  DiagnosticsLogger,
+  Networks,
+  Stream,
+  StreamState,
+  StreamStateHolder,
+  StreamUtils
+} from '@ceramicnetwork/common'
+import { AbstractStore } from './abstract-store.js'
+import { StoreWrapperInterface } from './store-wrapper-interface.js'
+
+export class StateStore extends AbstractStore<StreamID, StreamState> {
+  #logger: DiagnosticsLogger
+
+  constructor(logger: DiagnosticsLogger) {
+    super()
+    this.#logger = logger
+  }
+
+  /**
+   * Gets network name
+   */
+  get networkName(): string {
+    return this.store.networkName
+  }
+
+  getKey(object: StreamID): string {
+    return object.toString()
+  }
+
+  serialize(value: StreamState): any {
+    return StreamUtils.serializeState(value)
+  }
+
+  deserialize(serialized: any): StreamState {
+    return StreamUtils.deserializeState(serialized)
+  }
+
+  /**
+   * Pin stream
+   * @param stream - Stream instance
+   */
+  async saveFromStream(stream: Stream): Promise<void> {
```

Comment:
`Stream` implements `StreamStateHolder`. `saveFromStreamStateHolder` could be removed if here you replace `stream: Stream` with `stream: StreamStateHolder`.

Reply:
I added two different methods for saving from stream and from stream state holder to minimise the risk of regression (as I replaced the `LevelStateStore` and `S3StateStore` with a single class and one of the old classes had `save(stream: Stream)` and the other had `save(stateHolder: StreamStateHolder)`).

Do you feel confident that if I unify this, nothing will break :) ?

Reply:
`save(stateHolder: StreamStateHolder)` is a more abstract version of `save(stream: Stream)`, so my expectation is that it should not break anything.